### PR TITLE
[WIP] feat(identity): oauth2-proxy support

### PR DESCRIPTION
**Pull request by shahargl**
close https://github.com/keephq/keep/issues/1203

### CODE REVIEW

File: keep/api/config.py [hunk 1]
```diff
@@ -16,15 +16,22 @@
...
     logger.info("Keep server starting")
-    
+
     migrate_db()
 
     # Create single tenant if it doesn't exist
     if AUTH_TYPE in [
-        AuthenticationType.SINGLE_TENANT.value,
-        AuthenticationType.NO_AUTH.value,
+        IdentityManagerTypes.DB.value,
```
Comment (talboren) [inline keep/api/config.py:24]: is this right? "DB" is the new single tenant mode?

File: keep/api/core/db_on_start.py [hunk 2]
```diff
@@ -42,7 +42,7 @@
 engine = create_db_engine()
 
 
-def try_create_single_tenant(tenant_id: str) -> None:
+def try_create_single_tenant(tenant_id: str, create_default_user=True) -> None:
```
Comment (talboren) [inline keep/api/core/db_on_start.py:45]: why is this needed?

**Review by talboren — DISMISSED** [review]
lgtm, elegant

File: keep/api/config.py (hunk 1 above)
Comment (shahargl) [reply keep/api/config.py:24]: yep - with single_tenant the users are being stored on the db (also authentication) so the "provider" here is db

File: keep/api/core/db_on_start.py (hunk 2 above)
Comment (shahargl) [reply keep/api/core/db_on_start.py:45]: cuz with oauth2proxy we don't want default user

**Comment by talboren** [comment]
Closing for now as this is not ready yet.
